File: app/src/main/java/com/srlee/dlx/DLXNode.java
package com.srlee.dlx;

class DLXNode extends LL2DNode {
	public final DLXColumn columnHeader;
	private final int rowIndex;
	
	DLXNode(final DLXColumn columnHeader, final int rowIndex) {
		this.rowIndex = rowIndex;
		this.columnHeader = columnHeader;
		columnHeader.U.D = this;
		U = columnHeader.U;
		D = columnHeader;
		columnHeader.U = this;
		columnHeader.IncSize();
	}
	
	DLXColumn GetColumn() {
		return columnHeader;
	}
	
	int GetRowIdx() {
		return rowIndex;
	}
}

File: gauguin-core/src/main/kotlin/org/piepmeyer/gauguin/creation/dlx/DLXNode.kt
package org.piepmeyer.gauguin.creation.dlx

internal class DLXNode(val column: DLXColumn, val row: Int) : LL2DNode() {

    init {
        column.up!!.down = this
        up = column.up
        down = column
        column.up = this
        column.incrementSize()
    }
}
